sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java

```diff
@@ -68,19 +72,48 @@ public XmlSensor(FileSystem fileSystem, CheckFactory checkFactory, FileLinesCont
 
   @Override
   public void execute(SensorContext context) {
+    List<InputFile> inputFiles = new ArrayList<>();
+    fileSystem.inputFiles(mainFilesPredicate).forEach(inputFiles::add);
+
+    if (inputFiles.isEmpty()) {
+      return;
+    }
+
+    ProgressReport progressReport = new ProgressReport("Report about progress of XML analyzer", TimeUnit.SECONDS.toMillis(10));
+    progressReport.start(inputFiles.stream().map(InputFile::toString).collect(Collectors.toList()));
 
-    for (InputFile inputFile : fileSystem.inputFiles(mainFilesPredicate)) {
-      try {
-        XmlFile xmlFile = XmlFile.create(inputFile);
-        LineCounter.analyse(context, fileLinesContextFactory, xmlFile);
-        runChecks(context, xmlFile);
-        saveSyntaxHighlighting(context, XMLHighlighting.highlight(xmlFile), inputFile);
-      } catch (Exception e) {
-        processParseException(e, context, inputFile);
+    boolean successfullyCompleted = false;
+    boolean cancelled = false;
```

Comment:
It seems to me that these two booleans are always linked. Couldn't we use only one of them?

Reply:
indeed I can probably get rid of `successfullyCompleted`